src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,254 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import abc
+
+NEXT_BREAKING_CHANGE_RELEASE = '2.61.0'
+
+
+def _next_breaking_change_version():
+    return NEXT_BREAKING_CHANGE_RELEASE
+
+
+# pylint: disable=too-few-public-methods
+class TargetVersion(abc.ABC):
+    @abc.abstractmethod
+    def __str__(self):
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def version(self):
+        raise NotImplementedError()
+
+
+# pylint: disable=too-few-public-methods
+class NextBreakingChangeWindow(TargetVersion):
+    def __str__(self):
+        return f'in next breaking change release({_next_breaking_change_version()})'
+
+    def version(self):
+        return _next_breaking_change_version()
+
+
+# pylint: disable=too-few-public-methods
+class ExactVersion(TargetVersion):
+    def __init__(self, version):
+        self.version = version
+
+    def __str__(self):
+        return f'in {self.version}'
+
+    def version(self):
+        return self.version()
+
+
+# pylint: disable=too-few-public-methods
+class UnspecificVersion(TargetVersion):
+    def __str__(self):
+        return 'in future'
+
+    def version(self):
+        return None
+
+
+class BreakingChange(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def message(self):
+        pass
+
+    @property
+    @abc.abstractmethod
+    def target_version(self):
+        pass
+
+
+class Remove(BreakingChange):
```

Comment:
a little concern about BreakingChange related class naming. It might be a little short and common, like `Remove`, `Rename`, and may cause name conflicts easily. 

For me, adding a common prefix for these classes' name would be more clear and specific, like `AzCLIRenameLog`, `AzCLIRemoveLog` or something else? ( I add `Log` here cause these classes seem to be used to log all kinds of breaking changes infos, you might have different considerations.)

Reply:
These BreakingChange related classes would only be used in `_breaking_change.py` files. 
Here’s an example of how they could be used:

```python
from azure.cli.core.breaking_change import upcoming_breaking_changes, DefaultChange, NextBreakingChangeWindow

# The default value of an argument would be changed in a future release.
upcoming_breaking_changes['bar foo'] = DefaultChange('--type', 'TypeA', 'TypeB', target_version=NextBreakingChangeWindow())
```

I think a short class name could be concise and is not easy to conflict in this context. It is reasonable to add a prefix. But I couldn't find a suitable suffix and am concerned to be too long.